### PR TITLE
✨   amp-ad adocean: added support for master-slave and nonpersonalized ads

### PR DESCRIPTION
**Pull request by mmuras**
- Implement master-slave support
- Implement support for non personalized ads

### CODE REVIEW

**Comment by torch2424** [comment]
Triaging to @zhouyx 😄 , feel free to re-assign

File: ads/adocean.js [hunk 1]
```diff
@@ -178,15 +317,35 @@ export function adocean(global, data) {
...
+                   || global.context.initialConsentState === CONSENT_POLICY_STATE.UNKNOWN_NOT_REQUIRED);
+
   writeScript(global, adoUrl, () => {
-    setupAdoConfig(mode, global);
+    setupAdoConfig(mode, global, consent);
     setupPreview(global, data);
 
-    appendPlacement(mode, global, data);
+    if (masterId) {
+      if (global['ado']['features'] && global['ado']['features']['passback']) {
```
Comment (zhouyx) [inline ads/adocean.js]: is `global['ado']` always guaranteed to be defined?

File: ads/adocean.md [hunk 2]
```diff
@@ -18,27 +18,56 @@ limitations under the License.
...
+### Optional parameters
 
 - `data-ao-mode` - sync|buffered - processing mode
 - `data-ao-preview` - livepreview configuration id
-- `data-ao-keys` - additional configuration, see adserver documentation
-- `data-ao-vars` - additional configuration, see adserver documentation
+- `data-ao-keys` - additional configuration, see adserver documentation, do not define different values for slaves within one master
+- `data-ao-vars` - additional configuration, see adserver documentation, do not define different values for slaves within one master
 - `data-ao-clusters` - additional configuration,see adserver documentation
+- `data-ao-master` - unique id of master placement
```
Comment (zhouyx) [inline ads/adocean.md]: Could you please add a section to describe how user consent is being processed. You can refer to our [example doc](https://github.com/ampproject/amphtml/blob/master/ads/_ping_.md#user-consent-integration) for reference. Thank you

**Comment by mmuras** [comment]
Because of problems with CLA, I've asked colleague to open another pull request: https://github.com/ampproject/amphtml/pull/21328 and I'm closing this one. @zhouyx sorry for the late response, and please continue there. Over the next week I'll prepare the appropriate changes, according to your comments.

File: ads/adocean.js (hunk 1 above)
Comment (mmuras) [reply ads/adocean.js]: It's defined inside external script, which can failed to load. I'll add additional check for that.

Comment (mmuras) [reply ads/adocean.js]: It's fixed in https://github.com/ampproject/amphtml/pull/21328

File: ads/adocean.md (hunk 2 above)
Comment (mmuras) [reply ads/adocean.md]: It's added in https://github.com/ampproject/amphtml/pull/21328

**Pull request closed without merging**